pkg/apis/options/options.go

```diff
@@ -58,7 +58,7 @@ type Options struct {
 	SkipJwtBearerTokens   bool     `flag:"skip-jwt-bearer-tokens" cfg:"skip_jwt_bearer_tokens"`
 	ExtraJwtIssuers       []string `flag:"extra-jwt-issuers" cfg:"extra_jwt_issuers"`
 	SkipProviderButton    bool     `flag:"skip-provider-button" cfg:"skip_provider_button"`
-	SSLInsecureSkipVerify bool     `flag:"ssl-insecure-skip-verify" cfg:"ssl_insecure_skip_verify"`
+	SSLInsecureSkipVerify bool     `cfg:",internal"`
```

Comment:
Why are you changing this flag? It is used for the upstream communication as well. Which we shouldn't touch as part of this PR.

Reply:
Tbh I can't completely remember the motivation behind moving the `flag` to the legacy_options. Probably with the intention to keep it closer to the other configs being added and thinking overlap would cause issues.

Moving things into the `ProviderTLS` struct seems to remove the last reference to this field in `Options` though.